crates/engine/src/prepare/trusted_documents.rs

```diff
@@ -79,26 +82,57 @@ impl<'ctx, R: Runtime> PrepareContext<'ctx, R> {
                             operation_name,
                             document: Cow::Borrowed(document),
                         },
-                        document_or_future: wrap_document(document),
+                        content: wrap_document(document),
+                    })
+                } else if let Some(query) = request.query.as_deref().filter(|query| !query.is_empty()) {
+                    // Just an inline document.
+                    let Some(client_name) = client_name else {
+                        return Err(missing_client_name_error());
+                    };
+                    let hash = sha2::Sha256::digest(query.as_bytes());
```

Comment:
why sha256 rather than blake3? Do we want sha256 specifically or a cryptographic hash?
I guess the `dbg!` underneath can be removed?

Reply:
Ah we assume the doc_id is a hex(sha256(query))

Reply:
It's because that's what apollo persisted queries use. We have to take a guess on the document id, basically, in this branch. It's best effort.

Reply:
The dbg should go, obviously, good catch.